frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-resource.ts

```diff
@@ -402,7 +394,9 @@ type OptionsList = {
   requestInit: RequestInit;
 };
 
-type K8sListResource = (options: OptionsList) => Promise<Response>;
+type K8sListResource = (
+  options: OptionsList,
+) => Promise<K8sResourceCommon[] | (K8sResourceCommon & { items: K8sResourceCommon[] })>;
```

Comment:
Any reason not to use generics here, too?

Reply:
No particular reason I suppose... the new functions that are coming in @vojtechszocs's work in the Core SDK will replace all of this, so whatever I do here is more of a stop-gap... but I guess it might be more useful to use generics. I'll do the update.